eng/packages/http-client-csharp/generator/Azure.Generator/src/AzureOutputLibrary.cs

```diff
@@ -76,7 +76,18 @@ private Dictionary<string, OperationSet> CategorizeClients()
 
         /// <inheritdoc/>
         // TODO: generate resources and collections
-        protected override TypeProvider[] BuildTypeProviders() => [.. base.BuildTypeProviders(), new RequestContextExtensionsDefinition()];
+        protected override TypeProvider[] BuildTypeProviders()
+        {
+            if (AzureClientPlugin.Instance.IsAzureArm.Value == true)
+            {
+                var armOperation = new MgmtLongRunningOperationProvider(false);
+                AzureClientPlugin.Instance.AddTypeToKeep(armOperation.Name);
```

Comment:
Shouldn't this happen in `Configure`?

Reply:
Actually, they should be removed once they are reference in the Resource operation implementation, added TODO for them.
Not quite following `happen in Configure`. You mean we take `MgmtLongRunningOperationProvider` out from outputLIbraryand add them in Configure? Do you mean `AddTypeToKeep` should only be invoked in Configure?

Reply:
Yeah I think it is better to invoke the various AddXXX methods from the plugin Configure override.

Reply:
https://github.com/microsoft/typespec/blob/2c01003b02c6cf981765b0687088df1f2f71ede9/packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/ScmOutputLibrary.cs#L68
There is a usage of AddTypeToKeep outside Configure in MGC as well, it's just more convenient to add the TypeProvider after creating. Otherwise, you need to get it afterwards from OutputLibrary in Configure.

Reply:
Anyway, the usages of `AddTypeToKeep` for `MgmtLongRunningOperationProvider` will be removed once they are used for Resource operation implementation. I think leaving TODOs here would be enough.